### DISCUSSION TITLE
Automerge in AzureDevOps issue. canMergeReason undefined

### DISCUSSION THREAD

**Discussion by namrog84** [renovatebot/renovate#12750] [Request Help]
Selfhost on AzureDevOps. (via just running `npx renovate`) on some build machines. 

DEBUG: Branch status green (repository=React/Feedback, branch=renovate/jest-monorepo)
...
DEBUG: PR is configured for automerge (repository=React/Feedback, branch=renovate/jest-monorepo)
DEBUG: PR is not ready for merge (repository=React/Feedback, branch=renovate/jest-monorepo)
       "canMergeReason": undefined
...
DEBUG: resolveBranchStatus(branchName=renovate/jest-monorepo, ignoreTests=false) (repository=React/Feedback, branch=renovate/jest-monorepo)

It has been approved and all status are good/green. e.g. https://i.imgur.com/wLLjUhr.png

No changes to the repo has occurred.  Yet I've re-run renovatebot several times.

I am wondering if perhaps I need to ignoreTests:true because ADO's PR build gates/green status isn't the same as github's green status? 

Otherwise I am not sure why it is not ready for merge?  I feel like everything is good to go on it.

**Comment by namrog84** [comment]
So I just tried with ignoreTests:true and it automerged 1. Will need do more testing to make sure it can't merge a thing it shouldnt, but I assume I am in a scenario that ignoreTests:true is for because we don't have CI.Tests, but PR build gate/tests instead.  So it might be good for us to go that way?

**Comment by rarkins** [comment]
`canMergeReason` is a github term so it's not the root cause and something you can disregard here.

By searching for the log string I can find that you hit this logic: https://github.com/renovatebot/renovate/blob/7801ae7c1652426de10557555fc7c71dbfd47f71/lib/workers/pr/automerge.ts#L45-L49

This means that `pr.canMerge` must be `false`. Searching through `lib/platform/azure/*` I can't find anywhere where `pr.canMerge` is set, so maybe it's a concept we don't support for Azure? Cc @JamieMagee in case you recall

Setting `ignoreTests` to `true` would indeed be a way around that but in general we don't recommend it as you may end up merging failed updates.

**Reply by namrog84** [reply]
I do see this code here.

https://github.com/renovatebot/renovate/blob/main/lib/platform/azure/index.ts#L396

In azure, perhaps we shouldn't be using `autoMerge`'s default path but Azure's platform "Auto Complete".
Which is what I normally do to my own PRs. I am finding some mention of `platformAutomerge` or `azureAutoComplete`


Aha!
https://docs.renovatebot.com/configuration-options/#platformautomerge
Maybe this is what I need instead?
> platformAutomerge will configure PRs to be merged after all (if any) branch policies have been met. This option is available for Azure, GitHub and GitLab

Yeah I think this is what I overlooked.

**Reply by rarkins** [reply]
Yes, I recommend enabling platformAutomerge

**Reply by namrog84** [reply]
Aha! that did it! Looks likes its all working as I need it too now.  Excellent! 
https://i.imgur.com/FT71UL1.png
